File: skills/conductor/examples/agents/typescript/openai-agents-bridge.ts
/**
 * OpenAI Agents SDK on Conductor (verbatim from the server UI quickstart).
 *
 * Install: npm install @io-orkes/conductor-javascript @openai/agents   (pin from the SDK README you fetched)
 * Env:     CONDUCTOR_SERVER_URL (+ CONDUCTOR_AUTH_KEY/SECRET if required). The TS SDK still reads
 *          CONDUCTOR_SERVER_URL / CONDUCTOR_AUTH_KEY / CONDUCTOR_AUTH_SECRET env (SDK 4.x); tunables via the AgentRuntime options object.
 *
 * Execution model: COMPILED -- LLM turns run on the server (OPENAI_API_KEY on the server). Any
 * function tools become worker tasks that need runtime.serve(agent).
 */
import { Agent, setTracingDisabled } from "@openai/agents";
import { AgentRuntime } from "@io-orkes/conductor-javascript/agents";

setTracingDisabled(true);
const agent = new Agent({
  name: "openai_greeter",
  model: "gpt-4o-mini",
  instructions: "You are friendly and concise.",
});

const runtime = new AgentRuntime();
try {
  const result = await runtime.run(agent, "Share a durable execution fact.");
  result.printResult();
} finally {
  await runtime.shutdown();
}

File: skills/conductor/examples/agents/python/openai_agents_bridge.py
"""OpenAI Agents SDK on Conductor: change one import.

Install: pip install 'conductor-python[openai-agents]'  (pin versions from the SDK README you fetched)
Env:     CONDUCTOR_SERVER_URL (+ CONDUCTOR_AUTH_KEY/SECRET if required)

Execution model: COMPILED -- the server maps the OpenAI Agent onto a Conductor graph, so LLM turns
run on the server (OPENAI_API_KEY belongs on the server, not in this process). Any @function_tool
tools become worker tasks and need a serve() process (AgentRuntime().serve(agent)) to execute.
Alternative to the drop-in Runner: pass the OpenAI `Agent` straight to AgentRuntime().run(agent, prompt).
"""
from conductor.ai import Runner
from agents import Agent

agent = Agent(
    name="haiku_agent",
    model="gpt-4o-mini",
    instructions="Answer only in haiku.",
)

result = Runner.run_sync(agent, "Write about durable execution.")
print(result.final_output)
print(result.execution_id)
